python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -177,39 +199,98 @@ async def read_loop(self) -> None:
                 # spectrum analyzer.
                 await self.write(self.set_freq_start_cmd)
                 await self.write(self.set_freq_stop_cmd)
-            while self.connected or self.simulation_mode != 0:
+            while self.connected:
                 timestamp = utils.current_tai()
-                if self.simulation_mode == 0:
-                    await self.write(self.query_trace_data_cmd)
-                    assert self.client is not None  # make mypy happy
-                    read_bytes = await asyncio.wait_for(
-                        self.client.readuntil(TERMINATOR),
-                        timeout=self.config.read_timeout,
+                await self.write(self.query_trace_data_cmd)
+                assert self.client is not None  # make mypy happy
+                read_bytes = await asyncio.wait_for(
+                    self.client.readuntil(TERMINATOR),
+                    timeout=self.config.read_timeout,
+                )
+                raw_data = read_bytes.decode().strip()
+                raw_data_items = raw_data.split(",")
+                data = [float(i.strip()) for i in raw_data_items]
+                if (
+                    len(data) < EXPECTED_NUMBER_OF_DATA_POINTS
+                    and self.number_of_truncated_data_read < MAX_NUM_TRUNCATED_DATA
```

Comment:
Please reset self.number_of_truncated_data_read to 0 in `connect` or `disconnect`, so you always start with a clean slate.

But I am uncomfortable with the logic. I think your desire is primarily to allow the *first* read to be truncated. Instead, you allow up to one read to be truncated at any point. If you really want to do the former then I suggest something like this (and no EXPECTED_NUMBER_OF_DATA_POINTS constant):

Set the following in `connect` and `__init__`:
```
self.read_data = False  # set this in connect
```
Then in the read loop:
```
if len(data) < EXPECTED_NUMBER_OF_DATA_POINTS and self.read_data:
    raise ...
 self.read_data = True
 ```

Reply:
This still needs to be done, for the same reason as resetting self.num_consecutive_read_timeouts.

Though I think it is probably better to only allow one truncated read: the first one. Truncated data after one successful read is pretty scary and has an entirely different cause than having the first read unusable.